components/scream/src/physics/rrtmgp/rrtmgp_inputs_initializer.cpp

```diff
@@ -0,0 +1,184 @@
+        count += m_fields.count("sw_flux_up");
+        count += m_fields.count("sw_flux_dn");
+        count += m_fields.count("sw_flux_dn_dir");
+        count += m_fields.count("lw_flux_up");
+        count += m_fields.count("lw_flux_dn");
+  
+        if (count==0) {
+          return;
+        }
+  
+        EKAT_REQUIRE_MSG(
+            count==18,
+            "Error! RRTMGPInputsInitializer is expected to init\n"
+            "       pmid, pint, tmid, tint, col_dry, gas_vmr, sfc_alb_dir, sfc_alb_dif,\n"
+            "       mu0, lwp, iwp, rel, rei,\n"
+            "       sw_flux_up, sw_flux_dn, sw_flux_dn_dir, lw_flux_up, lw_flux_dn,\n"
+            "       but only " + std::to_string(count) + " of those have been found.\n"
+            "       Please, check the atmosphere processes you are using,\n"
+            "       and make sure they agree on who's initializing each field.\n"
+        );
+  
+        // Get device views
+        auto d_pmid = m_fields.at("pmid").get_view();
+        auto d_pint = m_fields.at("pint").get_view();
+        auto d_tmid = m_fields.at("tmid").get_view();
+        auto d_tint = m_fields.at("tint").get_view();
+        auto d_col_dry = m_fields.at("col_dry").get_view();
+        auto d_gas_vmr = m_fields.at("gas_vmr").get_view();
+        auto d_sfc_alb_dir = m_fields.at("sfc_alb_dir").get_view();
+        auto d_sfc_alb_dif = m_fields.at("sfc_alb_dif").get_view();
+        auto d_mu0 = m_fields.at("mu0").get_view();
+        auto d_lwp = m_fields.at("lwp").get_view();
+        auto d_iwp = m_fields.at("iwp").get_view();
+        auto d_rel = m_fields.at("rel").get_view();
+        auto d_rei = m_fields.at("rei").get_view();
+        auto d_sw_flux_up = m_fields.at("sw_flux_up").get_view();
+        auto d_sw_flux_dn = m_fields.at("sw_flux_dn").get_view();
+        auto d_sw_flux_dn_dir = m_fields.at("sw_flux_dn_dir").get_view();
+        auto d_lw_flux_up = m_fields.at("lw_flux_up").get_view();
+        auto d_lw_flux_dn = m_fields.at("lw_flux_dn").get_view();
+  
+        // RRTMGP initialization routine needs YAKL Fortran-style arrays, which
+        // we can map from the C-style Kokkos views. Note that this assumes that
+        // the Kokkos views are LayoutRight, and that dimension ordering is
+        // opposite what RRTMGP expects (that is, (nx,ny,nz) -> (nz,ny,nx))
+        // Read in dummy Garand atmosphere; if this were an actual model simulation,
+        // these would be passed as inputs to the driver
+        // NOTE: set ncol to size of col_flx dimension in the input file. This is so
+        // that we can compare to the reference data provided in that file. Note that
+        // this will copy the first column of the input data (the first profile) ncol
+        // times. We will then fill some fraction of these columns with clouds for
+        // the test problem.
+        int ngas =  8;
+        int ncol = 128;
```

Comment:
This appears to be a hard-coded value.  What would happen if we ran with a different number of columns or layers?

Reply:
The test would fail :-) This is hard-coded for now just for the purpose of the standalone test. I will work on making this flexible.